### PR TITLE
[FIX] website_event: filters

### PR DESCRIPTION
**Pull request by morm-odoo**
Prior to this PR, there were displays and design issues with the both desktop and mobile filters on the `website_event` list of events.

- Prior to the first commit(686d4e04c960538b484717dd908dc50d85c274a3):
The `countries dropdown` filter was still a `li` on desktop.
The spacing between the `dropdowns` was also wrong.
The `countries filter` was showing after the `searchbar`, so the template
position is modified.
`IsActive` value is set to ease classes conditions on the dropdowns links.

- Prior to the second commit(c40dcc2d29e5d0c97dbdc3db154f9153ed0868ea):
 The `countries` filter was not appearing in the off-canvas filters on mobile.
The `date` filter was also appearing all the time in the off-canvas, even when it was not supposed to.

- Prior to the 3rd commit(2d348d58cfa227fa84dc1581d80e81ec58e4cd23):
The position of the `<li>` and `t-foreach` was not correct.
They needed to be swapped, because we had single `<li>` with multiples `<a>` inside, instead of multiples `<li>` with a single `<a>`.

- Prior to the 4th commit(10116a1aa4a430eb86099cc463d4376294e662ad):
The design of all the `off-canvas filters` was wrong.
It was `list-group` with a `dropdown-item` class on the elements inside.
Also there was a wrong `border-top` class on the `accordion button`, causing display issues on `hover.
`Aria-control` was also missing from the `categories filters`.

- Prior to the 5th commit(37cff6b9bfeb940f0dfc3d729d547518ea6a360a):
 There was no `title` attributes in the accordions of the filters.
They are also added in the `website_event.pot` file
Checkboxes and radio buttons are also added to increases the usability.

task-3626957

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by anso-odoo — APPROVED** [review]
Hello, nice job, LGTM 👍

**Review by reth-odoo — COMMENTED** [review]
Hi :)

Ok for me to duplicate some stuff as we don't want to modify more templates than necessary.

Just not sure we want to fully remove the existing code or just make it invisible, for stability. Either way if someone had a custom override it will break in some way so..
@tde-banana-odoo ?

File: addons/website_event/views/event_templates_list.xml [hunk 1]
```diff
@@ -204,7 +205,42 @@
...
+            <h2 class="accordion-header">
+                <button class="accordion-button border-top collapsed"
+                    type="button"
+                    data-bs-toggle="collapse"
+                    data-bs-target=".o_wevent_offcanvas_country"
+                    aria-expanded="false">
+                    Countries
+                </button>
+            </h2>
+            <div class="o_wevent_offcanvas_country accordion-collapse collapse">
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:221]: Can this have the same class name as the div it's replacing? Makes overriding easier and not option-dependent.

File: addons/website_event/views/event_templates_list.xml [hunk 2]
```diff
@@ -204,7 +205,42 @@
...
-        </li>
+        </div>
+    </xpath>
+    <xpath expr="//span[hasclass('o_wevent_offcanvas_location')]" position="replace">
+        <div class="accordion-item">
+            <h2 class="accordion-header">
+                <button class="accordion-button border-top collapsed"
+                    type="button"
+                    data-bs-toggle="collapse"
+                    data-bs-target=".o_wevent_offcanvas_country"
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:216]: Same here then, of course (alternatively could change `o_wevent_offcanvas_location` to `o_wevent_offcanvas_country`)

**Review by reth-odoo — COMMENTED** [review]
Baring any commend from tde, in the end I would say it's better for it to fail explicitly if people had incompatible overrides.

Just a small comment

**Comment by reth-odoo** [comment]
For those joining us here's why we need to change the xpath selector:
image

The other change is just moving from always-display to the optional override, so the option actually does something

**Comment by morm-odoo** [comment]
@stefanorigano Could you check please ? :)

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @morm-odoo , thanks for your work 👍 

I've found some minors about accessibility... as you may know we decided to focus more on it in order to make the product more inclusive, especially the frontend.

There are some details about the the rendering of the countries' badge, but my main remark is the offcanvas `list-group` content organization. The approach you used doesn't follow the BS default architecture and the visual result is affected.

Read my comments, I'd suggest to have a look on how we do that in `/shop` or other offcanvas 😉

Thanks!

| /events (this PR) | /shop |
|--------|--------|
| image | image |

File: addons/website_event/views/event_templates_list.xml [hunk 3]
```diff
@@ -177,14 +152,40 @@
...
+                Date
+            </button>
+        </h2>
+        <div class="o_wevent_offcanvas_date accordion-collapse collapse">
+            <div class="accordion-body pt-0">
+                <ul class="list-group list-group-flush">
+                    <li class="list-group-item">
+                        <t t-foreach="dates" t-as="date">
+                            <t t-if="date[3] or (date[0] in ('old','upcoming','all'))">
+                                <a t-att-href="keep('/event', date=date[0])" t-attf-class="dropdown-item d-flex align-items-center justify-content-between #{searches.get('date') == date[0] and 'active'}">
```
[the remark this answers is not in the document]
Comment (morm-odoo) [reply addons/website_event/views/event_templates_list.xml]: Will be fixed on master on a following task (task-3675802)

File: addons/website_event/views/event_templates_list.xml [hunk 4]
```diff
@@ -177,14 +152,40 @@
...
+                data-bs-target=".o_wevent_offcanvas_date"
+                aria-expanded="false">
+                Date
+            </button>
+        </h2>
+        <div class="o_wevent_offcanvas_date accordion-collapse collapse">
+            <div class="accordion-body pt-0">
+                <ul class="list-group list-group-flush">
+                    <li class="list-group-item">
+                        <t t-foreach="dates" t-as="date">
```
[the remark this answers is not in the document]
Comment (morm-odoo) [reply addons/website_event/views/event_templates_list.xml:171]: Will be fixed on a following task on master (task-3675802)

**Comment by morm-odoo** [comment]
@stefanorigano

File: addons/website_event/views/event_templates_list.xml [hunk 5]
```diff
@@ -160,51 +135,120 @@
...
             </a>
             <div class="dropdown-menu">
                 <t t-foreach="countries" t-as="country">
                     <t t-if="country['country_id']">
-                        <a t-att-href="keep('/event', country=country['country_id'][0])" t-attf-class="dropdown-item d-flex align-items-center justify-content-between #{searches.get('country') == str(country['country_id'] and country['country_id'][0]) and 'active'}">
+                        <t t-set="isActive" t-value="searches.get('country') == str(country['country_id'] and country['country_id'][0])"/>
+                        <a role="button" t-att-href="keep('/event', country=country['country_id'][0])" t-attf-class="dropdown-item d-flex align-items-center justify-content-between #{isActive and 'active'}" t-att-title="country['country_id'][1]">
                             <t t-out="country['country_id'][1]"/>
-                            <span t-out="country['country_id_count']" class="badge rounded-pill text-bg-primary ms-auto"/>
+                            <span t-out="country['country_id_count']" t-attf-class="badge rounded-pill #{isActive and 'text-bg-light' or 'text-bg-primary'} ms-3"/>
```
Comment (stefanorigano) [inline addons/website_event/views/event_templates_list.xml:201]: convention: text utility classes after margins ones

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @morm-odoo , thanks.

I know that you plan to fix on `master` the issues that this PR will introduce, but I don't see the point of merging in production templates that are visually broken, especially since this PR aims to fix design issues.

If this was a "one-line-fix" I would have agreed, but we reached 100+ lines already and mostly all of these are **new** lines; we should code them right from the start, without the need of follow-up tasks on master.

Luckily this component is not directly editable by the user, thus we can exclude website editor conflicts with production databases.

Some notes for you in the code.
I'll push a fixup commit for you to check, thanks.

**Comment by stefanorigano** [comment]
@morm-odoo , pushed 97d39349a33c156e15d992413704ed6d85bb1b50 on top for you to check. Thanks

| this PR | --fixup |
|--------|--------|
| image | image |

File: addons/website_event/views/event_templates_list.xml [hunk 6]
```diff
@@ -99,47 +99,24 @@
...
                                         <ul class="list-group list-group-flush">
-                                            <li t-if="category.is_published and category.tag_ids and any(tag.color for tag in category.tag_ids)" class="list-group-item">
-                                                <t t-foreach="category.tag_ids" t-as="tag">
-                                                    <a t-if="tag.color"
-                                                        t-att-href="'/event?%s' % keep_query('*', tags=str((search_tags - tag).ids if tag in search_tags else (tag | search_tags).ids))"
-                                                        t-attf-class="dropdown-item d-flex align-items-center justify-content-between #{'active' if tag in search_tags else ''}">
-                                                        <t t-out="tag.name"/>
+                                            <t t-foreach="category.tag_ids" t-as="tag">
+                                                <li t-if="category.is_published and category.tag_ids and any(tag.color for tag in category.tag_ids) and tag.color" class="list-group-item border-0 px-0">
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:103]: Indeed good fix, but should be/remain in its own commit

Can move most of the t-if before the t-foreach
```suggestion
                                            <t t-if="category.is_published and category.tag_ids and any(tag.color for tag in category.tag_ids)" t-foreach="category.tag_ids" t-as="tag">
                                                <li t-if="tag.color" class="list-group-item border-0 px-0">
```

File: addons/website_event/views/event_templates_list.xml [hunk 7]
```diff
@@ -160,51 +137,127 @@
 <!-- Filter - Date -->
 <template id="event_time" inherit_id="website_event.index_topbar" name="Filter by Date">
     <xpath expr="//div[hasclass('o_wevent_search')]" position="before">
-        <div class="dropdown d-none d-lg-block me-2 my-1">
-            <a href="#" role="button" class="btn btn-light dropdown-toggle" data-bs-toggle="dropdown">
+        <div class="dropdown d-none d-lg-block">
+            <a href="#" role="button" class="btn btn-light dropdown-toggle" data-bs-toggle="dropdown" title="Filter by Date">
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:141]: May need to add this in the .pot

Not sure these changes are relevant so maybe omit them? Or have them in a separate commit at least.
(i.e. don't fixup and keep a separate commit with a description of the changes)

File: addons/website_event/views/event_templates_list.xml [hunk 8]
```diff
@@ -160,51 +137,127 @@
...
                 <t t-if="current_date" t-out="current_date"/>
                 <t t-else="">Upcoming Events</t>
             </a>
             <div class="dropdown-menu">
                 <t t-foreach="dates" t-as="date">
                     <t t-if="date[3] or (date[0] in ('old','upcoming','all'))">
-                        <a t-att-href="keep('/event', date=date[0])" t-attf-class="dropdown-item d-flex align-items-center justify-content-between #{searches.get('date') == date[0] and 'active'}">
+                        <t t-set="isActive" t-value="searches.get('date') == date[0]"/>
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:148]: Nitpick but it would be better to follow the python scheme in python templates
```suggestion
                        <t t-set="is_active" t-value="searches.get('date') == date[0]"/>
```

same everywhere else

**Review by reth-odoo — COMMENTED** [review]
Will leave you to it but overall if we can keep the fixes separate that'd be good :)

**Comment by stefanorigano** [comment]
Hello @reth-odoo , thanks for your review and support 🙂
> if we can keep the fixes separate that'd be good :)

Sure. I've pushed on top it's to speed-up things and to clarify Morgane the points I wanted to tickle 😉
At the current state the PR is not ready to be merged OFC.
I'll let @morm-odoo handle commits organization and their descriptions according to your requirements. 

Thanks again

File: addons/website_event/views/event_templates_list.xml [hunk 9]
```diff
@@ -88,26 +88,29 @@
...
                             <div class="accordion-item">
                                 <h2 class="accordion-header">
-                                    <button class="accordion-button border-top collapsed"
+                                    <button class="accordion-button collapsed"
                                         type="button"
                                         data-bs-toggle="collapse"
                                         t-att-data-bs-target="'.o_wevent_offcanvas_cat_%s' % category.name"
-                                        aria-expanded="false">
+                                        aria-expanded="false"
+                                        t-attf-aria-controls="o_wevent_offcanvas_cat_#{category.name}">
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:96]: Seems like this makes more sense above `t-att-data-bs-target`, also this doesn't seem to be mentioned in the commit

File: addons/website_event/views/event_templates_list.xml [hunk 10]
```diff
@@ -113,33 +113,8 @@
                                 </div>
                             </div>
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:117]: commit title sugg: "respect filter options on mobile"

File: addons/website_event/views/event_templates_list.xml [hunk 11]
```diff
@@ -99,9 +99,9 @@
                                 <div t-attf-class="o_wevent_offcanvas_cat_#{category.name} accordion-collapse collapse">
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml]: commit title sugg: put foreach before list-item nodes

File: addons/website_event/views/event_templates_list.xml [hunk 12]
```diff
@@ -99,9 +99,9 @@
...
                                     <div class="accordion-body pt-0">
                                         <ul class="list-group list-group-flush">
-                                            <li t-if="category.is_published and category.tag_ids and any(tag.color for tag in category.tag_ids)" class="list-group-item">
-                                                <t t-foreach="category.tag_ids" t-as="tag">
-                                                    <a t-if="tag.color"
+                                            <t t-if="category.is_published and category.tag_ids and any(tag.color for tag in category.tag_ids)" t-foreach="category.tag_ids" t-as="tag">
+                                                <li t-if="tag.color" class="list-group-item border-0 px-0">
+                                                    <a t-att-href="'/event?%s' % keep_query('*', tags=str((search_tags - tag).ids if tag in search_tags else (tag | search_tags).ids))"
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:105]: Seems like this is a duplicate of the following line?

File: addons/website_event/views/event_templates_list.xml [hunk 13]
```diff
@@ -88,26 +88,29 @@
                         <t t-foreach="categories" t-as="category">
```
Comment (reth-odoo) [inline addons/website_event/views/event_templates_list.xml:88]: commit title sugg: tweak mobile filter design

**Review by reth-odoo — COMMENTED** [review]
Seems ok code-wise I think

A few suggestions as commit titles are normally sentences from the guidelines, and `off-canvas` isn't super intuitive wording

Will give it a spin too just to be sure

**Review by reth-odoo — COMMENTED** [review]
@tde-banana-odoo @jeh-odoo hello

I'm told this needs to be merged yesterday (but for friday actually) so I'm pinging already if one of you wants to give this one a look

Checked that it was ok functionally, there's some suboptimal changes for stable but this is a bigram request and a lot of the changes are for truly broken UI so :shrug: 

As I understand @stefanorigano is ok with the design?

Don't have much more to say on this one as far as the code goes. Cheers

File: addons/website_event/views/event_templates_list.xml (hunk 12 above)
Comment (morm-odoo) [reply addons/website_event/views/event_templates_list.xml:105]: I don't understand, can you explain?

**Review by stefanorigano — APPROVED** [review]
> As I understand @stefanorigano is ok with the design?

Design LGTM, thanks 👍

File: addons/website_event/views/event_templates_list.xml (hunk 8 above)
Comment (reth-odoo) [reply addons/website_event/views/event_templates_list.xml:149]: not addressed yet

**Review by reth-odoo — COMMENTED** [review]
Went over it again, didn't find anything to say except I still have an unresolved comment :)

I feel like the first commit could be merged with the "tweak mobile filter design" one if that makes sense, as it doesn't have much to stand on its own.

Once resolved, nothing else to say on my part.
Thanks for the work

**Comment by jeh-odoo** [comment]
@robodoo rebase-ff r+
